### ISSUE TITLE
Temperature charge limitation does not work

### ISSUE THREAD

**Issue by pecheneg11** [lich4/ChargeLimiter#67]
Hello! Starting from version 1.5, the temperature charge limitation does not work. Installing the previous version solves the problem. 1.6 also doesn't work. iOS 15.6.1, Air 5

**Comment by lich4** [comment]
If you have 1.4- version installed with TrollStore, you should uninstall it first, and install version 1.4.1/1.5/1.6

**Comment by AnthoPakPak** [comment]
For me the temperature limit never worked. I'm checking after each update but it keeps stopping charging whereas my temperature is way lower than the threshold. For example, had it setup at 45C max and it stopped while the temp was 32C. Let me know if you need any info to help troubleshoot :)

**Comment by lich4** [comment]
> For me the temperature limit never worked. I'm checking after each update but it keeps stopping charging whereas my temperature is way lower than the threshold. For example, had it setup at 45C max and it stopped while the temp was 32C. Let me know if you need any info to help troubleshoot :)
> 
> (And I never had the trollstore version)

I don't know why it not work for you, but it do work for most users,(I have a small team help me to test before each release). You can join the Telegram channel, and submit your log(/var/root/aldente.log) and 5min chart image, I will help you fix this issue.

**Comment by lich4** [comment]
The temperature maybe affected by environment, so CL couldn't get full control of it.
If temperature control is not work, you need to describe it in detail. For example, the max and min temperature threshhold, 5min chart image, the log(/var/root/aldente.log), and which part is not work?both the max and min threshhold? or only the min?

**Comment by pecheneg11** [comment]
> If you have 1.4- version installed with TrollStore, you need to uninstall it first, and install version 1.4.1/1.5/1.6

Unfortunately, this won't help me. Since I installed it via Sileo, not via TrollStore. I just want the battery to stop charging when the battery temperature reaches 34 degrees Celsius. But this only works in version 1.4.1. Now I will try to install version 1.6 via the TrollStore. I will test and write about the result

**Comment by AnthoPakPak** [comment]
> I don't know why it not work for you, but it do work for most users,(I have a small group help me to test before each release). You can join the Telegram group, and submit your log(/var/root/aldente.log) and 5min chart image, I will help you fix this issue.

I'll reenable the setting and provide you logs later. Maybe I'm understanding the settings incorrectly… I have it set up like this:

![IMG_9352B3E7E3ED-1](https://github.com/lich4/ChargeLimiter/assets/16854916/176b6d80-9589-43a7-a3c5-97f830154377)

With this setup, I expect it to stop charging if temp > 45°C or battery > 79%. And last time I tried with this setup, it wasn't charging whereas my device was at 41% and reported battery temperature was at 33°C. From my understanding, it should charge, unless either the temp or the battery is reached. Each time, disabling the the temperature control toggle allowed my device to start charging again.

> which part is not work?both the max and min threshhold? or only the min?

I'd say this is only the max threshold which isn't working, but I never had a chance to try the min one.

But now that you talk about the min threshold I've took another look, and I wonder if it's not related to that. "Start charging if temp <= 10°C" as in my current settings might be what causes it to not charge, isn't it? That may be the actual cause of the issue 🤔

**Comment by lich4** [comment]
> > I don't know why it not work for you, but it do work for most users,(I have a small group help me to test before each release). You can join the Telegram group, and submit your log(/var/root/aldente.log) and 5min chart image, I will help you fix this issue.
> 
> I'll reenable the setting and provide you logs later. Maybe I'm understanding the settings incorrectly… I have it set up like this:
> 
> ![IMG_9352B3E7E3ED-1](https://private-user-images.githubusercontent.com/16854916/329869475-176b6d80-9589-43a7-a3c5-97f830154377.jpeg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTU1NjEyNTEsIm5iZiI6MTcxNTU2MDk1MSwicGF0aCI6Ii8xNjg1NDkxNi8zMjk4Njk0NzUtMTc2YjZkODAtOTU4OS00M2E3LWEzYzUtOTdmODMwMTU0Mzc3LmpwZWc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTEzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUxM1QwMDQyMzFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mZjZjMjkwNDJkYjZlNGRiYjBhN2QyNzI2MDU3NDRlMmYyOWU4ZTZkOWM3ZjI3M2I5OTA5MzQxNGFjMDI3NTY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.JuipquW500iQHT7uI6mnyYQizfgf0ALLB8Y4PnYymB8)
> 
> With this setup, I expect it to stop charging if temp > 45°C or battery > 79%. And last time I tried with this setup, it wasn't charging whereas my device was at 41% and reported battery temperature was at 33°C. From my understanding, it should charge, unless either the temp or the battery is reached. Each time, disabling the temperature control toggle allowed my device to start charging again.
> 
> > which part is not work?both the max and min threshhold? or only the min?
> 
> I'd say this is only the max threshold which isn't working, but I never had a chance to try the min one.
> 
> But now that you talk about the min threshold I've took another look, and I wonder if it's not related to that. "Start charging if temp <= 10°C" as in my current settings might be what causes it to not charge, isn't it? That may be the actual cause of the issue 🤔

CL will only charge/discharge for certain conditions, and stay at the status if none of the conditions match. For example, if the max is 45°C and the min is 10°C, and the battery have reached 45°C ever, CL will stop charging. And if the battery never reach 10°C, it will not recover charge, and temperature will drop if not charge. It can be easily traced in "5min chart" and in the log.

**Comment by AnthoPakPak** [comment]
So after a few days I can confirm I was understanding the settings incorrectly. I've now changed the "Start charging Temperature (°C) <=" setting to its max value, and temperature control works fine for me.
I was only interested in stopping charging when temp is too high. It seems having "Start charging Temperature (°C) <=" to 10°C actually waited for the temp to be <= 10°C before charging again, which obviously rarely happens in normal conditions.
Thanks for your support and this great tweak/app! 👍

**Comment by lich4** [comment]
CL will not recover charge automatically, so "Start charging Temperature" must be set correctly to resume charging if temperature control enabled

**Comment by pecheneg11** [comment]
It still doesn't work for me. When I installed version 1.6 via the TrollStore, I set the test temperature values close to the current of my battery and the CL worked. But after that, it stopped working again. I installed the version from Sileo and tested it for a while. I can say that it works and does not work, for unknown reasons. Once I managed to look at the chart for 5 minutes and for some reason everything was fine for a while. I'm going back to version 1.4.1.
